script.js

```diff
@@ -48,5 +48,55 @@ const photos = [
         type: 'animals',
     },
 ];
+const gallery = document.getElementById('gallery');
+const loadMoreBtn = document.getElementById('loadMore');
+let currentIndex = 0;
+const photosPerLoad = 6;
 
-console.log('hello');
\ No newline at end of file
+function renderPhotos(filterType = 'all', limit = true) {
+    gallery.innerHTML = '';
+    let filteredPhotos = filterType === 'all' ? photos : photos.filter(photo => photo.type === filterType);
+
+    const photosToShow = limit ? filteredPhotos.slice(0, currentIndex + photosPerLoad) : filteredPhotos;
+    
+    photosToShow.forEach(photo => {
+        const img = document.createElement('img');
+        img.src = photo.url;
+        gallery.appendChild(img);
+    });
+    
+    if (filteredPhotos.length > photosToShow.length) {
+        loadMoreBtn.style.display = 'block';
+    } else {
+        loadMoreBtn.style.display = 'none';
+    }
+}
+
+loadMoreBtn.addEventListener('click', () => {
+    currentIndex += photosPerLoad;
+    renderPhotos('all', true);
+});
+
+
+document.getElementById('all').addEventListener('click', () => {
+    currentIndex = 0;
+    renderPhotos('all');
+});
+
+document.getElementById('nature').addEventListener('click', () => {
+    currentIndex = 0;
+    renderPhotos('nature');
+});
+
+document.getElementById('city').addEventListener('click', () => {
+    currentIndex = 0;
+    renderPhotos('city');
+});
+
+document.getElementById('animals').addEventListener('click', () => {
+    currentIndex = 0;
+    renderPhotos('animals');
+});
```

Comment:
To clean up your code, a suggestion would be to select all the buttons in one array, and then use a for loop to create a generic click event function for them all.